src/DataValues/QuantityValue.php

```diff
@@ -130,6 +130,26 @@ public function __unserialize( array $data ): void {
 		$this->__construct( $amount, $unit, $upperBound, $lowerBound );
 	}
 
+	public function getSerializationForHash(): string {
+		// mimic a legacy serialization of __serialize() (amount + unit + upperBound + lowerBound)
+		$amountSerialization = method_exists( $this->amount, 'getSerializationForHash' )
```

Comment:
Note: `DecimalValue` is itself a `DataValueObject` subclass, so it inherits `getSerializationForHash()` from there. The inherited version is fine, so we don’t need to touch `DecimalValue.php`.

(And the `method_exists()` check should in theory ensure that this still works with pre-3.1.0 DataValues versions, but I haven’t tested that.)